ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardSyncStep.java

```diff
@@ -36,23 +37,47 @@ public BackwardSyncStep(final BackwardsSyncContext context, final BackwardChain
   }
 
   @Override
-  public CompletableFuture<Void> executeStep() {
-    return CompletableFuture.supplyAsync(this::earliestUnprocessedHash)
+  public CompletableFuture<Void> executeOneStep() {
+    return CompletableFuture.runAsync(this::waitForTTD)
+        .thenApply(this::earliestUnprocessedHash)
         .thenCompose(this::requestHeader)
         .thenApply(this::saveHeader)
         .thenApply(this::possibleMerge)
         .thenCompose(this::possiblyMoreBackwardSteps);
   }
 
+  private void waitForTTD() {
+    while (!context.isOnTTD()) {
+      LOG.info("Did not reach TTD yet, falling asleep...");
+      try {
+        Thread.sleep(5000);
```

Comment:
why not using scheduleFutureTaskWithFixedDelay instead of a sleep. It seems we are not using a lot of Thread.sleep in besu